### PR TITLE
Create gdrive.py and fix incomplete zip download recovery

### LINKED ISSUE #47: If you cancel initial download when running as script, it does not clean itself up correctly
The original zip download fails problematically if you stop it partway through the download.

It leaves two side effects:

```
allofplos/allofplos_xml/allofplos_xml.zip
zip_info.txt
```

Both of these should probably be removed if they are not completed appropriately (e.g., if you hit ⌃-C). That probably means being put in a context manager.

### PR DESCRIPTION
**Pull request by eseiver**
This closes #47.

This fix was buried in the corpus_class branch and accidentally was never applied to master. Two main things:
– Google Drive functionality is moved over to a new file, `gdrive.py`
– zipfile checks that the allofplos xml zip is valid before extracting. if not (i.e. the download was interrupted), it's deleted and the download is started over.

Also now a download bar for the extraction process, which takes some time. Confirmed working as expected and successfully downloaded corpus from scratch, even after interrupting the zip download.

### CODE REVIEW

**Comment by mpacer** [comment]
I kinda would like to avoid introducing another new top level module.

Can you explain what all of the different modules involved in downloading and managing the corpus files are? From what I understand the corpus class is actually unrelated to this functionality, right?

File: allofplos/gdrive.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+import tarfile
+from zipfile import ZipFile, BadZipFile
+
+import requests
+from tqdm import tqdm
+
+from . import get_corpus_dir
+
+# Variables needed
+zip_id = '0B_JDnoghFeEKLTlJT09IckMwOFk'
```
Comment (mpacer) [inline allofplos/gdrive.py]: Global variables should be in ALL_SNAKE_CAPS by PEP8 convention… I hate it but it does make it clearer when something is a global variable.

File: allofplos/gdrive.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+metadata_id = '0B_JDnoghFeEKQUhKWXBOVy1aTlU'
```
Comment (mpacer) [inline allofplos/gdrive.py]: what's the difference between these ids?

File: allofplos/gdrive.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+local_zip = 'allofplos_xml.zip'
+zip_metadata = 'zip_info.txt'
+time_formatting = "%Y_%b_%d_%Hh%Mm%Ss"
```
Comment (mpacer) [inline allofplos/gdrive.py]: Is this the same time_formatting you use elsewhere? is anything else expecting this time_formatting to be followed?

File: allofplos/gdrive.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+min_files_for_valid_corpus = 200000
```
Comment (mpacer) [inline allofplos/gdrive.py]: is this arbitrary?

File: allofplos/gdrive.py [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+test_zip_id = '12VomS72LdTI3aYn4cphYAShv13turbX3'
```
Comment (mpacer) [inline allofplos/gdrive.py]: how is this related to the zip_id?

File: allofplos/gdrive.py [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+local_test_zip = 'sample_corpus.zip'
+gdrive_url = "https://docs.google.com/uc?export=download"
+
+
+def download_file_from_google_drive(id, filename, destination=None,
```
Comment (mpacer) [inline allofplos/gdrive.py]: why is it called destination not directory?

File: allofplos/gdrive.py (hunk 2 above)
Comment (eseiver) [reply allofplos/gdrive.py]: these point at 2 different files on google drive. the zip_id is the actual zip, and the metadata_id is the text file with zip file size and date info

File: allofplos/gdrive.py (hunk 4 above)
Comment (eseiver) [reply allofplos/gdrive.py]: A cut-off to avoid too much scraping journal pages for new articles. You could conceivably download one article and then run `plos_corpus` and download the other 227000 manually. This assures that they have enough articles from the zip that it won't be too costly otherwise, though technically the number is abitrary

**Comment by eseiver** [comment]
@mpacer eventually hope to deprecate plos_corpus. Would you want to put them all in a corpus folder?

File: allofplos/gdrive.py (hunk 3 above)
Comment (eseiver) [reply allofplos/gdrive.py]: @mpacer this is how times are used for the zip file naming in the private repo. Do you have a preferred format? Not used anywhere else for e.g. pubdates, which are only year month and day

**Comment by eseiver** [comment]
Superseded by #78.
